chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -407,84 +432,102 @@ func (b *BitcoindNotifier) handleRelevantTx(tx chain.RelevantTx, bestHeight int3
-	txConf, err := b.confDetailsFromTxIndex(txid)
-	if err != nil {
-		return nil, err
-	}
+	// We'll first attempt to retrieve the transaction using the node's
+	// txindex.
+	txConf, txStatus, err := b.confDetailsFromTxIndex(txid)
+
+	// We'll then check the status of the transaction lookup returned to
+	// determine whether we should proceed with any fallback methods.
+	switch {
+	// The transaction was found within the node's txindex.
+	case txStatus == txFoundMempool:
+
+	// The transaction was found within the node's txindex.
+	case txStatus == txFoundIndex:
 
-	if txConf != nil {
-		return txConf, nil
+	// The transaction was not found within the node's mempool or txindex.
+	case txStatus == txNotFoundIndex && err == nil:
+
+	// We failed to look up the transaction within the node's mempool or
+	// txindex, so we'll proceed to scan the chain manually.
+	default:
+		return b.confDetailsManually(txid, heightHint, currentHeight)
 	}
 
-	// If the backend node's transaction index is not enabled, then we'll
-	// fall back to manually scanning the chain's blocks, looking for the
-	// block where the transaction was included in.
-	return b.confDetailsManually(txid, heightHint, currentHeight)
+	return txConf, txStatus, nil
 }
 
 // confDetailsFromTxIndex looks up whether a transaction is already included
 // in a block in the active chain by using the backend node's transaction index.
 // If the transaction is found, its confirmation details are returned.
 // Otherwise, nil is returned.
 func (b *BitcoindNotifier) confDetailsFromTxIndex(txid *chainhash.Hash,
-) (*chainntnfs.TxConfirmation, error) {
+) (*chainntnfs.TxConfirmation, txConfStatus, error) {
 
 	// If the transaction has some or all of its confirmations required,
 	// then we may be able to dispatch it immediately.
 	tx, err := b.chainConn.GetRawTransactionVerbose(txid)
 	if err != nil {
-		// Avoid returning an error if the transaction index is not
-		// enabled to proceed with fallback methods.
+		// If the transaction lookup was succesful, but it wasn't found
+		// within the index itself, then we can exit early. We'll also
+		// need to look at the error message returned as the error code
+		// is used for multiple errors.
+		txNotFoundErr := "No such mempool or blockchain transaction"
 		jsonErr, ok := err.(*btcjson.RPCError)
-		if !ok || jsonErr.Code != btcjson.ErrRPCNoTxInfo {
-			return nil, fmt.Errorf("unable to query for txid "+
-				"%v: %v", txid, err)
+		if ok && jsonErr.Code == btcjson.ErrRPCNoTxInfo &&
+			strings.Contains(jsonErr.Message, txNotFoundErr) {
+
+			return nil, txNotFoundIndex, nil
 		}
+
+		return nil, txNotFoundIndex, fmt.Errorf("unable to query for "+
+			"txid %v: %v", txid, err)
 	}
 
 	// Make sure we actually retrieved a transaction that is included in a
 	// block. Without this, we won't be able to retrieve its confirmation
 	// details.
 	if tx == nil || tx.BlockHash == "" {
-		return nil, nil
+		return nil, txFoundMempool, nil
```

Comment:
Is it in the mempool in this case? Update comment to explain.

Reply:
Doesn't the comment as is imply this though by it not being included in a block?